config/database.yml

```diff
@@ -1,23 +1,31 @@
+<% if ENV.key?("MYSQL_ROOT_PASSWORD") %>
```

Comment:
I totally agree. However, I've tried it, and while it is shorter, it seems to me to be less readable and clear, the logic not apparent at first glance. While not too important, it also ends up in a long line (101 chars) and won't have a nice place to put the comment about old behaviour. Looks like this (maybe there's a better way to do it?):

```erb
<% user = ENV.key?("MYSQL_ROOT_PASSWORD") ? "root" : ENV["MYSQL_USER"] %>
<% password = ENV.key?("MYSQL_ROOT_PASSWORD") ? ENV["MYSQL_ROOT_PASSWORD"] : ENV["MYSQL_PASSWORD"] %>
```